src/validatePrTitle.js

```diff
@@ -120,7 +120,35 @@ module.exports = async function validatePrTitle(
   }
 
   if (subjectPattern) {
-    const match = result.subject.match(new RegExp(subjectPattern));
+    // eslint-disable-next-line no-inner-declarations
+
+    // Define a whitelist of allowed special characters
+    const allowedSpecialChars = [
+      '.',
+      '*',
+      '+',
+      '?',
+      '^',
+      '$',
+      '{',
+      '}',
+      '(',
+      ')',
+      '|',
+      '[',
+      ']',
+      '\\'
+    ];
+
+    // Escape all special characters that are not in the whitelist
+    const sanitizedPattern = subjectPattern.replace(
+      /([.*+?^${}()|[\]\\])/g,
+      (match) => (allowedSpecialChars.includes(match) ? match : `\\${match}`)
+    );
```

Comment:
Can you provide an example of a regex that would be changed with this logic? From what I can tell the regex matches all chars that are allowed, therefore I'm wondering what would change based on this?

Reply:
Related to this, I found these npm libraries:
- https://www.npmjs.com/package/safe-regex
- https://www.npmjs.com/package/safe-regex2

Note that both claim that:

> WARNING: This module has both false positives and false negatives. It is not meant as a full checker, but it detect basic cases.